src/views/List/List.jsx

```diff
@@ -63,18 +86,65 @@ export default function List({ token }) {
     });
   };
 
+  //onChange handler for search input
+  const filterList = (e) => {
+    const { value } = e.target;
+    setSearchInput(value);
+    //the original data set is filtered so we always search from all the list items
+    let searchResults = data.filter((listItem) => {
+      return listItem.name.toLowerCase().includes(searchInput.toLowerCase());
+    });
+    //if search input is empty (the user hasnt typed anything or they removed their input), set the data back to the original list
+    if (value === '') {
+      setCopyOfData(data);
+      setToggleErr(true);
+    } else {
+      setCopyOfData(searchResults);
+      setToggleErr(false);
+    }
+  };
+
   return (
     <section>
       <div className="div">
+        {data.length > 1 ? (
+          <label htmlFor="search">
+            Search List:
+            <input
+              title="search your list"
+              aria-label="enter your search term"
+              type="search"
+              name="search"
+              id="search"
+              onChange={filterList}
+              value={searchInput || ''}
+              placeholder="e.g. potatoes"
+            />
+          </label>
+        ) : null}
         {data.length ? (
           <ul>
-            {data.map((listItem, index) => {
-              console.log(listItem);
+            {searchError ? (
+              <p
+                role="alert"
+                id="search-err"
+                className="search-error"
+                style={{
+                  outline: '3px dashed red',
+                }}
+              >
+                {searchError}
+              </p>
+            ) : null}
+            {copyOfData.map((listItem, index) => {
               const { name, isActive } = listItem;
               return (
                 <li key={index}>
                   {' '}
                   <input
+                    aria-invalid={toggleErr}
```

Comment:
I love how your pair demoed this with the video in your PR and showed us the compatibility with assistive tech! ✨